crates/binstalk-downloader/src/remote.rs

```diff
@@ -107,6 +110,11 @@ impl Client {
                 .https_only(true)
                 .tcp_nodelay(false);
 
+            #[cfg(feature = "trust-dns")]
+            {
+                builder = builder.dns_resolver(Arc::new(DefaultResolver::default()));
```

Comment:
It's really unfortunate that we need another `Arc` here.

Unfortunately I can't think of any solution for this.

Reply:
`reqwest` does the same:
https://github.com/seanmonstar/reqwest/blob/17c893ffc0d3832d61cb1c0cf278340b7e95557e/src/async_impl/client.rs#L273

Reply:
Ok, given that it's not very expensive, I would have to accept it.
Thanks for reading `reqwest` to dig out the information for me!